### PR TITLE
WP-CLI: Reattach Reconnect-Related Events After a Successful Reconnection

### PR DESCRIPTION
**Pull request by jblz**
## Description

Currently, when a connection to the back end is broken and the client successfully reconnects a first time, the event listeners are not reset to allow subsequent reconnections to succeed.  This change re-binds the listeners, so additional reconnections per run are possible.

Also, in case the `cli` receives an `exit` signal, it will exit accordingly to the code and message received.

This change also adds some logging so we can monitor these situations and further improve the experience.

### Testing reattach reconnect-related events
1. Check out PR.
2. Run npm run build
3. Run ./dist/bin/vip-wp long-running-command
4. Simulate a network issue
5. Wait for the until to reconnect
6. Simulate another network issue
7. Wait for the until to reconnect
8. The command should resume/reconnect multiple times.

### Testing exit code

1. Check out PR.
2. Run npm run build
3. Run ./dist/bin/vip-wp long-running-command
4. Deploy a new version of the site
5. An error message should appear
6. Run `echo $?`
7. The output should be `101`

### CODE REVIEW

File: src/bin/vip-wp.js [hunk 1]
```diff
@@ -199,6 +202,55 @@ const launchCommandAndGetStreams = async ( { guid, inputToken, offset = 0 } ) =>
 	return { stdinStream, stdoutStream, socket };
 };
 
+const bindReconnectEvents = ( { cliCommand, inputToken, subShellRl, commonTrackingParams, isSubShell } ) => {
+	currentJob.socket.io.on( 'reconnect', async () => {
+		debug( '-------- reconnect' );
+		rollbar.info( 'WP-CLI socket.io.on( \'reconnect\' )', { custom: { code: 'wp-cli-on-reconnect', commandGuid: cliCommand.guid } } );
```
Comment (jblz) [inline src/bin/vip-wp.js:200]: I thought some rollbar events might be handy here to make sure we're seeing these come through to the client.  What do you think, @luismulinari?

Comment (luismulinari) [reply src/bin/vip-wp.js:200]: It is perfect!

**Comment by jblz** [comment]
@luismulinari -- Thanks for taking this over.  It's working well for me, but I can't approve the PR as the author.

Consider this a :+1:  from me!

**Review by luismulinari — APPROVED** [review]
LGTM! 🚢 

Thanks, Jeff!

**Pull request merged**